src/runner/bootstrapper.js

```diff
@@ -77,14 +77,29 @@ export default class Bootstrapper {
         var browsers           = await Promise.all(this.browsers.map(Bootstrapper._convertBrowserAliasToBrowserInfo));
         var browserConnections = browsers.map(browser => this._createConnectionFromBrowserInfo(browser));
 
-        try {
-            await this._waitBrowserConnectionsReady(browserConnections);
-        }
-        catch (err) {
+        function closeLocalConnections () {
             browserConnections.forEach(bc => {
                 if (bc instanceof LocalBrowserConnection)
                     bc.close();
             });
+        }
+
+        var disconnectedUserAgents = browserConnections.filter(bc => bc.disconnected).map(bc => bc.userAgent);
+
+        if (disconnectedUserAgents.length) {
+            closeLocalConnections();
```

Comment:
I haven't figured out that we will need to close local connections. From this perspective previous code was much better. So, let's roll it back. Sorry for that.